modules/basicfuncs/str-funcs.c

```diff
@@ -433,3 +433,75 @@ tf_string_padding(LogMessage *msg, gint argc, GString *argv[], GString *result)
 }
 
 TEMPLATE_FUNCTION_SIMPLE(tf_string_padding);
+
+typedef struct _TFBinaryState
+{
+  TFSimpleFuncState super;
+  GString *octets;
+} TFBinaryState;
+
+static void
+tf_binary_call(LogTemplateFunction *self, gpointer s, const LogTemplateInvokeArgs *args, GString *result)
+{
+  TFBinaryState *state = (TFBinaryState *) s;
+
+  g_string_append_len(result, state->octets->str, state->octets->len);
+}
+
+static gboolean
+tf_binary_prepare(LogTemplateFunction *self, gpointer s, LogTemplate *parent, gint argc, gchar *argv[], GError **error)
+{
+  TFBinaryState *state = (TFBinaryState *) s;
+  GString *octets = g_string_new("");
+
+  if (argc < 2)
+    {
+      g_set_error(error, LOG_TEMPLATE_ERROR, LOG_TEMPLATE_ERROR_COMPILE,
+                  "$(binary) Incorrect parameters, usage $(binary <number> <number> ...)");
+      goto error;
+    }
+
+  for (gint i = 1; i < argc; i++)
+    {
+      gint64 number;
+
+      gchar *token = argv[i];
+      if (!parse_number(token, &number))
+        {
+          g_set_error(error, LOG_TEMPLATE_ERROR, LOG_TEMPLATE_ERROR_COMPILE,
+                      "$(binary) template function requires list of dec/hex/oct numbers as arguments, unable to parse %s as a number",
+                      token);
+          goto error;
+        }
+      if (number > 0xFF)
+        {
+          g_set_error(error, LOG_TEMPLATE_ERROR, LOG_TEMPLATE_ERROR_COMPILE,
+                      "$(binary) template function only supports 8 bit values as characters, %" G_GUINT64_FORMAT " is above 255", number);
+          goto error;
+        }
+      g_string_append_c(octets, (guint8) number);
```

Comment:
This should be casted to gchar instead of guint8.

Reply:
done